apache/superset
# feat: Keep searching result

### SUMMARY
The feature change involves retaining search filter results across SQL Lab tabs by integrating search text into the Redux state.

Under the previous system, users would encounter the resetting of their search results each time they switched to a different SQL Lab tab and back. This process required users to repeatedly enter their search filters, resulting in a disrupted workflow.

This enhancement directly tackles this inconvenience. Upon rendering SQL Lab, it now utilizes the search text stored in Redux state as the filter value. This ensures that no matter how many tabs a user switches between, the search filter results remain persistent, no longer disappearing upon tab switching.

### BEFORE SCREENSHOTS OR ANIMATED GIF
https://github.com/apache/superset/assets/76474644/79a9d872-87c3-49d9-9e9e-01b958c4f86c

### AFTER SCREENSHOTS OR ANIMATED GIF
https://github.com/apache/superset/assets/76474644/801704af-a891-4405-8749-1064a0980210

### TESTING INSTRUCTIONS
- Navigate to the SQLLab page
- Execute Query
- Apply a filter on the resulting output table using the search box
- Navigate to other  SQL editor tab
- Return to the initial SQL editor tab
- Notice that the previously input search text and filter results are still visible

### ADDITIONAL INFORMATION
